### PR TITLE
Validate registered_name field in deserialize_keras_object

### LINKED ISSUE #22703: [Bug] deserialize_keras_object accepts arbitrary registered_name values without validation
## Description
The `registered_name` field in `deserialize_keras_object` is accepted without validation. Arbitrary string values, including unrelated module paths, are allowed and do not raise any error.

## Reproduction Script

```python
import os
os.environ["KERAS_BACKEND"] = "tensorflow"

from keras.saving import deserialize_keras_object

# Case 1: arbitrary registered_name
config_1 = {
    "class_name": "Dense",
    "module": "keras.layers",
    "registered_name": "builtins.eval",
    "config": {"units": 1}
}

obj_1 = deserialize_keras_object(config_1)
print("Case 1 type:", type(obj_1))


# Case 2: another arbitrary value
config_2 = {
    "class_name": "Dense",
    "module": "keras.layers",
    "registered_name": "subprocess.Popen",
    "config": {"units": 1}
}

obj_2 = deserialize_keras_object(config_2)
print("Case 2 type:", type(obj_2))


# Case 3: empty string vs None
config_3 = {
    "class_name": "Dense",
    "module": "keras.layers",
    "registered_name": "",
    "config": {"units": 1}
}

config_4 = {
    "class_name": "Dense",
    "module": "keras.layers",
    "registered_name": None,
    "config": {"units": 1}
}

obj_3 = deserialize_keras_object(config_3)
obj_4 = deserialize_keras_object(config_4)

print("Case 3 type (empty string):", type(obj_3))
print("Case 4 type (None):", type(obj_4))
print("Same behavior (empty vs None):", type(obj_3) == type(obj_4))
````

## Observed behavior

* Arbitrary values for `registered_name` are accepted without validation
* Values such as `"builtins.eval"` and `"subprocess.Popen"` do not raise errors
* Empty string (`""`) and `None` are treated equivalently

## Expected behavior

* `registered_name` should be validated or restricted to known/registered values
* Invalid or unexpected values should raise an error or be rejected
* Behavior should be consistent and clearly defined

## Notes

The `registered_name` field currently has no effect on deserialization and is not validated. This may lead to confusion or inconsistent handling of configuration data.

### PR DESCRIPTION
**Pull request by bzantium**
## Description

Fixes #22703.

The `registered_name` field of a serialized object config was accepted without validation: arbitrary strings like `"builtins.eval"` or `"subprocess.Popen"` passed through silently, non-string types were accepted, and empty string vs. `None` were treated equivalently without being documented as such. The values were not actually dangerous (the existing `_retrieve_class_or_fn` lookup only resolves entries from the Keras registry / `custom_objects`, so unknown names fall through), but the field gave no feedback at all when it was meaningless.

This change adds a small validation step when reading the field:

- reject non-string, non-`None` values with `ValueError` (per the Keras style guide, user input validation uses `ValueError`);
- normalize empty string to `None` so the two behave the same;
- warn when an explicitly provided non-empty value neither resolves via `custom_objects` / `GLOBAL_CUSTOM_OBJECTS` / the active custom object scope nor equals `class_name` (the legacy default). In that case the field has no effect on deserialization, so the warning tells the user their value is being ignored and points at the expected `package>name` format.

The documented patterns (None for built-ins, `package>name` for registered classes, `registered_name == class_name` for legacy configs) continue to load without warnings, so existing saved models are unaffected.

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Comment by bzantium** [comment]
Re: the suggestion to also validate `registered_name` in the `module_objects` branch (lines 545-605):

I left that out deliberately — that path is entered internally by `keras.activations.get` and friends, and before recursing it repackages the config via `serialize_with_public_class` / `serialize_with_public_fn`, which produces a fresh `registered_name` (either `None` or the object's canonical registry name). The user's original `registered_name` is therefore discarded rather than acted on, so the silent-ignore concern from #22703 doesn't apply there in the same way.

Adding the same warning to that branch risks false-positive warnings for the internal callers (activations, etc.) and expanded scope beyond the reported issue. Happy to add it if a maintainer prefers, but I'd rather keep this PR narrow.

**Review by hertschuh — CHANGES_REQUESTED** [review]
This is not the approach. The whole point about registered names is that 3rd parties can use any name, and it doesn't have to match the class name, this is on purpose. Registered names are not used for keras classes and functions.

The only verification to do is that `registered_name` is `None` when the class or function that was resolved is a keras_* class or function.

I.e. verification should only kick in when the `registered_name` is ignored.

**Comment by hertschuh** [comment]
Fixed by https://github.com/keras-team/keras/pull/22729
